modules/webxr/webxr_interface.cpp

```diff
@@ -0,0 +1,328 @@
+						});
+						session.requestReferenceSpace('local').then(function (refSpace) {
+							Module['webxr_space'] = refSpace;
+
+							// Now that both Module.webxr_session and Module.webxr_space are set,
+							// our monkey-patched requestAnimationFrame() should kick in.
+							// When using the WebXR API Emulator, this gets picked up automatically,
+							// however, in the Oculus Browser on the Quest, we need to pause and
+							// resume the main loop.
+							Module.Library_Browser_mainLoop.pause();
+							window.setTimeout(function () { Module.Library_Browser_mainLoop.resume(); });
+						});
+					});
+				});
+			});
+		});
+	};
+
+	return true;
+};
+
+void WebXRInterface::uninitialize() {
+	if (initialized) {
+		ARVRServer *arvr_server = ARVRServer::get_singleton();
+		if (arvr_server != NULL) {
+			// no longer our primary interface
+			arvr_server->clear_primary_interface_if(this);
+		}
+
+		EM_ASM({
+			Module.webxr_session.end();
+
+			// Resetting these will switch back to window.requestAnimationFrame() for the main loop.
+			// @todo Do we need to pause/resume the main loop here too?
+			Module.webxr_session = null;
+			Module.webxr_space = null;
+			Module.webxr_frame = null;
+			Module.webxr_pose = null;
+		});
+
+		initialized = false;
+	};
+};
+
+bool WebXRInterface::_have_frame() {
+	return (bool) EM_ASM_INT({
+		return !!Module['webxr_session'] && !!Module['webxr_pose'];
+	});
+}
+
+Size2 WebXRInterface::get_render_targetsize() {
+	// @todo This method is untested!
+
+	Size2 target_size;
+
+	//EM_ASM({ console.log("get_render_targetsize()"); });
+
+	if (!initialized || !_have_frame()) {
+		// As a default, use half the window size.
+		target_size = OS::get_singleton()->get_window_size();
+		target_size.width /= 2.0;
+		return target_size;
+	}
+
+	int32_t* js_size = (int32_t*) EM_ASM_INT({
+		let glLayer = Module['webxr_frame'].session.renderState.baseLayer;
+		let view = Module['webxr_pose'].views[0];
+		let viewport = glLayer.getViewport(view);
+
+		//console.log("targetsize javascript viewport:");
+		//console.log(viewport.width + " " + viewport.height);
+
+		let buf = Module._malloc(2 * 4);
+		setValue(buf + 0, viewport.width, 'i32');
+		setValue(buf + 4, viewport.height, 'i32');
+		return buf;
+	});
+
+	target_size.width = js_size[0];
+	target_size.height = js_size[1];
+
+	free(js_size);
+
+	EM_ASM({
+		//console.log("targetsize C++ viewport:");
+		//console.log($0 + " " + $1);
+	}, target_size.width, target_size.height);
+
+	return target_size;
+};
+
+Transform WebXRInterface::get_transform_for_eye(ARVRInterface::Eyes p_eye, const Transform &p_cam_transform) {
+	Transform transform_for_eye;
+
+	// @todo In 3DOF where we only have rotation, we should take the position from the p_cam_transform, I think?
+
+	if (!initialized || !_have_frame()) {
+		transform_for_eye = p_cam_transform;
+		return transform_for_eye;
+	}
+
+	int view_index = (p_eye == ARVRInterface::EYE_RIGHT) ? 1 : 0;
```

Comment:
Note, if p_eye is EYE_MONO you need to return the center position between the two eyes. this is where the camera object is placed within the node tree. While not used for rendering there are other parts that use the nodes transform to position other object or query this info to get the players head position.

Reply:
Fixed in the latest commit!